models/skilledImpactRequestsModel.js

```diff
@@ -0,0 +1,72 @@
+const db = require('../database/dbConfig');
+
+const ProjectGoal = require('./projectGoalsModel.js');
+
+
+async function findSkilledImpactRequests(campaign_id) {
+    return db('project_goals')
+        .where({campaign_id})
+        .fullOuterJoin('skilled_impact_requests', 'skilled_impact_requests.id','project_goals.skilled_impact_request_id')
+        .select(
+           '*'
+        )
+        .then((returnedQuery)=>{
+            const skilledRequests = new Map();
```

Comment:
I would probably rewrite this whole block as a `.map()`. If that gets too complicated, feel free to write it with a `for` loop but maybe get rid of the `Map` and just go straight to pushing to an array.

Reply:
@dacruz21 This is taking some list of project goals (with all the info for its corresponding skilled impact request) and restructuring that data into the hierarchy we want. How would we rewrite this block with `map`?

The original way we did this was pulling all the skilled impact requests and then doing a `map` that involved pulling all the project goals associated with a skilled impact request, but we could just do a join at that point. I don't think it'd be a lot of extra queries since most campaigns won't have more than a few skilled impact requests, so that's an option.

Is that what you had in mind / thoughts?